warehouse/internal/loadfiles/loadfiles.go

```diff
@@ -192,155 +210,52 @@ func (lf *LoadFileGenerator) createFromStaging(ctx context.Context, job *model.U
 		}
 	}()
 
-	var g errgroup.Group
-
-	var sampleError error
-	for _, chunk := range lo.Chunk(toProcessStagingFiles, publishBatchSize) {
-		// td : add prefix to payload for s3 dest
-		var messages []stdjson.RawMessage
-		for _, stagingFile := range chunk {
-			payload := WorkerJobRequest{
-				UploadID:                     job.Upload.ID,
-				StagingFileID:                stagingFile.ID,
-				StagingFileLocation:          stagingFile.Location,
-				LoadFileType:                 job.Upload.LoadFileType,
-				SourceID:                     job.Warehouse.Source.ID,
-				SourceName:                   job.Warehouse.Source.Name,
-				DestinationID:                destID,
-				DestinationName:              job.Warehouse.Destination.Name,
-				DestinationType:              destType,
-				DestinationNamespace:         job.Warehouse.Namespace,
-				DestinationConfig:            job.Warehouse.Destination.Config,
-				WorkspaceID:                  job.Warehouse.Destination.WorkspaceID,
-				UniqueLoadGenID:              uniqueLoadGenID,
-				RudderStoragePrefix:          misc.GetRudderObjectStoragePrefix(),
-				UseRudderStorage:             job.Upload.UseRudderStorage,
-				StagingUseRudderStorage:      stagingFile.UseRudderStorage,
-				DestinationRevisionID:        job.Warehouse.Destination.RevisionID,
-				StagingDestinationRevisionID: stagingFile.DestinationRevisionID,
-			}
-			if revisionConfig, ok := destinationRevisionIDMap[stagingFile.DestinationRevisionID]; ok {
-				payload.StagingDestinationConfig = revisionConfig.Config
-			}
-			if slices.Contains(warehouseutils.TimeWindowDestinations, job.Warehouse.Type) {
-				payload.LoadFilePrefix = lf.GetLoadFilePrefix(stagingFile.TimeWindow, job.Warehouse)
-			}
-
-			payloadJSON, err := jsonrs.Marshal(payload)
-			if err != nil {
-				return 0, 0, fmt.Errorf("error marshalling payload: %w", err)
-			}
-			messages = append(messages, payloadJSON)
-		}
-
-		uploadSchemaJSON, err := jsonrs.Marshal(struct {
-			UploadSchema model.Schema
-		}{
-			UploadSchema: job.Upload.UploadSchema,
-		})
+	if !lf.AllowUploadV2JobCreation(job) {
+		err = lf.createUploadJobs(ctx, job, toProcessStagingFiles, publishBatchSize, uniqueLoadGenID)
 		if err != nil {
-			return 0, 0, fmt.Errorf("error marshalling upload schema: %w", err)
+			return 0, 0, err
```

Comment:
minor: better to add context and wrap the error. Makes debugging easier reading the error
```suggestion
			return 0, 0, fmt.Errorf("creating upload jobs: %w", err)
```